### ISSUE TITLE
Getting NoMethodError undefined method `include?' for nil:NilClass on add_roles

### ISSUE THREAD

**Issue by aristizabal95** [RolifyCommunity/rolify#446]
Hello, I've been having a weird issue with rolify since yesterday.
Whenever I try to add roles to a user, i get this 

`NoMethodError undefined method `include?' for nil:NilClass on add_roles`

pointing me to the file:line 
`/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rolify-5.1.0/lib/rolify/role.rb:17`

and using byebug to inspect the code, I realize that the reason is this

`17:    if !roles.include?(role)  # roles => nil`

I am also getting a similar issue when I try to do `user.has_role? :role_name` which I would guess is happening because of the same problem.

The weird thing is that before yesterday I was perfectly capable of adding and checking roles both from the app and from console without any problem, but now it seems I can't do any of this.

I've been trying to look around where exactly does this `roles` variable gets created, but can't seem to find out. I haven't touched anything inside the gem code, just looked upon to see if I can find the source of the problem.

Any help would be appreciated, thanks!

**Comment by aristizabal95** [comment]
I think I'm getting it now. So, experimenting inside the `add_role` scope, I found out that the variable

`roles`

actually means the `user.roles`, being user the User instance I am trying to add a role to.
So, that means that my users shouldn't have the roles attribute as `nil` ? Then what should they have instead? Any help would be appreciated, thanks.

**Comment by aristizabal95** [comment]
Got it! as soon as I realized the `roles` variable was the one form the user instance, I understood the error.

Since I wanted to do some validations inside my User model, I created a virtual attribute called `roles`.... so yeah, I was overriding the values of roles with an array that would be emptied after user creation. That is the reason I was getting nil form every user created. So, I changed this

`class User < ApplicationRecord`
`rolify`
`attr_accessor :roles`

To this

`class User < ApplicationRecord`
`rolify`
`attr_accessor :input_roles`

and used this virtual attribute instead for user validations.
Hope this helps to anyone that could have this error.
